### PR TITLE
Update gauges

### PR DESCRIPTION
**Pull request by rjleveque**
Major refactor in the process of adding gauges.

### CODE REVIEW

File: src/1d_nonuniform/utility_module.f90 [hunk 1]
```diff
@@ -0,0 +1,258 @@
+! Contains a number of useful functions
```
Comment (mandli) [inline src/1d_nonuniform/utility_module.f90:1]: Is the same as the 2D version?

**Review by mandli — COMMENTED** [review]
Is there a test for the Boussinesq code?  Also what time frame are you thinking in terms of merging this into GeoClaw proper?  I was thinking that it would be good to keep the history intact as much as possible for this.

**Comment by mandli** [comment]
@rjleveque Totally forgot about this.  Does this update to make the 1D gauges the same as 2D?

**Comment by rjleveque** [comment]
This is a year old by now so I should go through again and see if it's consistent with 2d, and also think about moving this into the main geoclaw repository.

**Comment by mandli** [comment]
How different is this from the 1D version in geoclaw now?

**Comment by rjleveque** [comment]
Right now I think there is only `geoclaw/src/1d/multilayer` in geoclaw master.

This version is on a nonuniform grid that can be chosen so the Courant number is close to 1 everywhere up to close to a shoreline, so it uses a much finer grid near shore than in the deep, and also include radial source terms and Boussinesq terms.   @mjberger and I are working on a 2d version of the Boussinesq solvers now (with some limited AMR capability), so I'll clean up this 1d code as part of that project.

**Comment by rjleveque** [comment]
I'll merge this for now, still working on cleaning this up to fold into clawpack/geoclaw

**Pull request merged**